java/ql/src/Security/CWE/CWE-643/XPathInjection.ql

```diff
@@ -11,10 +11,10 @@
  */
 
 import java
+import DataFlow::PathGraph
```

Comment:
This is really minor and not something I think we actually ever talked about, but we tend to put the `import DataFlow::PathGraph` last in the sequence of imports. As I said, it's really not very important (putting `import java` first is important, though, but that's another matter entirely), but the reason I mention this is to highlight that the `PathGraph` import is a different kind of import than the others: the other imports reference file-modules, whereas `PathGraph` is a module in scope and this import statement merely opens it to the top-level scope. (There might even have been discussions about whether this ought to have been a different keyword than `import`, such as perhaps `open`.)
Also, for completeness of information, importing an in-scope module mostly affects naming, i.e. we can `import DataFlow` if we want to write `Node` rather than `DataFlow::Node`, but in this case there's a separate reason: The `PathGraph` module contains two predicates annotated with `query` and putting such predicates in the top-level scope makes them part of the query on the same level as the `select` statement (and those `query predicate`s contain the graph that's required for a `path-problem` query).

Reply:
That's really interesting, thank you for taking the time to explain it with so much detail!

I moved the import down as you suggested (see b69be30) - now that I know the reason, it makes a lot of sense.